lib/spack/spack/package.py

```diff
@@ -2373,6 +2377,8 @@ def use_cray_compiler_names():
     os.environ['F77'] = 'ftn'
 
 
+# Note this is a feature required for external packages maintained by at least
```

Comment:
It's going to clutter the code to have a bunch of meta-conversation (this is used by n sites, etc.) in the comments.

I think notes like this should go in the tests for the feature.  Or, we should just add an explicit test to check that features like this (things needed by external projects) exist.  e.g. `test_externally_required_calls_are_present`.  This would keep the code cleaner.

Reply:
I added the comment to avoid the problem where a contributor removes a feature (without bothering to ask) that was not being explicitly used in the code base.  (Apparently it's happened on an actual code.)

Reply:
Yes, I'm just saying a better way to enforce that is to add a test.

Reply:
How about changing the current test to use a (new) built-in mock that uses the install function as Greg explained?  Or is that what you are suggesting?